# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/test/webgl-conf/checkout/conformance/offscreencanvas/context-lost-worker.html
# Commit: 8e0aea33f622
# Full Hash: 8e0aea33f622a76c5c8baa7dcf4485ffefd1662f
# Author: Kelsey Gilbert <kelsey.gilbert@mozilla.com>
# Date: 2024-07-09 09:45:28
# Regressor Bug: 1892542
# File Overlap Count: 1
# Description:
#   Bug 1892542 - Webgl e.g. createBuffer() made infallible. r=gfx-reviewers,webidl,saschanaz,lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D208050
# ==============================================================================

diff -r b953e0ccb68b -r 8e0aea33f622 dom/canvas/test/webgl-conf/checkout/conformance/offscreencanvas/context-lost-worker.html
--- a/dom/canvas/test/webgl-conf/checkout/conformance/offscreencanvas/context-lost-worker.html	Mon Jul 08 19:36:12 2024 +0300
+++ b/dom/canvas/test/webgl-conf/checkout/conformance/offscreencanvas/context-lost-worker.html	Mon Jul 08 17:24:12 2024 +0000
@@ -24,12 +24,11 @@
   var worker = new Worker('context-lost-worker.js');
   worker.postMessage("Start worker");
   worker.onmessage = function(e) {
-    if (e.data == "Test passed") {
-      testPassed("All tests have passed");
-    } else {
-      testFailed("Some tests failed");
+    const fn = e.data.fail ? testFailed : testPassed;
+    fn(e.data.msg);
+    if (e.data.finishTest) {
+      finishTest();
     }
-    finishTest();
     return;
   }
 }